src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+    ucc_mem_hints_t         hints;
+    ucc_mem_constraints_t   constraints;
+} ucc_mem_map_params_t;
+
+typedef struct  ucc_team_oob_coll {
+    int             (*allgather)(void *src_buf, void *recv_buf, size_t size,
+                                 void *allgather_info,  void **request);
+    ucc_status_t    (*req_test)(void *request);
+    ucc_status_t    (*req_free)(void *request);
+    uint32_t 	     participants;
+    void             *coll_info;
+}  ucc_team_oob_coll_t;
+
+typedef enum {
+    UCC_COLLECTIVE_POST_ORDERED = 0,
+    UCC_COLLECTIVE_POST_UNORDERED = 1
+} ucc_post_ordering_t;
+
+typedef enum {
+    UCC_COLLECTIVE_EP_RANGE_CONTIG = 0, 
+    UCC_COLLECTIVE_EP_RANGE_NONCONTIG = 1
+} ucc_ep_range_type_t;
+
+typedef struct ucc_team_params {
+    uint64_t                mask;
+    ucc_post_ordering_t     ordering;
+    uint64_t                outstanding_colls;
+    uint64_t                ep;
+    ucc_ep_range_type_t     ep_range;
+    ucc_coll_sync_type_t    sync_type;
+    ucc_team_oob_coll_t     oob_collective;
+    ucc_mem_map_params_t    mem_params;
+} ucc_team_params_t;
+
+typedef struct ucc_team_attribs {
+    uint64_t               mask;
+    ucc_post_ordering_t    ordering;
+    uint64_t               outstanding_colls;
+    uint64_t               ep;
+    ucc_ep_range_type_t    ep_range;
+    ucc_coll_sync_type_t   sync_type;
+    ucc_mem_map_params_t   mem_params;
+} ucc_team_attribs_t;
+
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @brief ucc_team_create_post is a nonblocking collective operation to create
+ * the team handle. It takes in parameters ucc_context_h, num_handles,
+ * ucc_team_params_t and returns a ucc_team_handle_h. The ucc_team_params_t
+ * provides user configuration to customize the team. The routine returns
+ * immediately after posting the operation with the new team handle. However,
+ * the team handle is not ready for posting the collective operation.
+ * ucc_team_create_test operation is used to learn the status of the new team
+ * handle. On error, the team handle will not be created and corresponding error
+ * code as defined by ucc_status_t is returned.
+ *
+ * @param  [in] contexts     Communication context abstracting the resources
+ * @param  [in] num_contexs  Number of context provided as input
+ * @param  [in] params       User defined configurations for the team
+ * @param  [out] ucc_team    Team handle created
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_post(
+              ucc_context_h context,
+              ucc_team_params_t team_params,
+              ucc_team_h *new_team);
+
+
+/**
+ *  @ingroup UCC_TEAM
+ *
+ *  @brief @ref ucc_team_create_test routines tests the status of team handle.
+ *  If required it can progress the communication but cannot block on the
+ *  communications.
+ *
+ *  @param  [in] ucc_team  Team handle to test
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_test();
```

Comment:
ucc_team_h as argument

Reply:
I debated about it. Given that our semantics require that only one outstanding team create operation at any given time, do we want it? I don’t have a strong opinion on it.

Reply:
I agree with Valentin, I think it's much more "readable" with the argument, and it will also let us make chage this semantic in the future if we want.

Reply:
As I mentioned, I don't have a stronger opinion. I agree, it is cleaner also.

Reply:
Added it.